src/Libraries/CoreNodes/String.cs

```diff
@@ -429,7 +430,11 @@ public static string Remove(string str, int startIndex, int? count = null)
 
             if (_count < 0)
             {
-                startIndex += _count;
+                // The input count is negative means the removal process is required operate from right to left.
+                // However, the removal process in this function always operates rightwards (left to right).
+                // Therefore, a conversion for start index needs to be done in order to change
+                // from leftwards removal (right to left) to rightwards removal (left to right).
+                startIndex = startIndex + _count + 1; 
```

Comment:
If the startIndex and Count are out of range, the exception of argumentOutOfRange is thrown, the string.remove node in the UI turn yellow with the meaningful error message for users.